authors/bruno-skvorc/polkadot/en_US/final.md

```diff
@@ -0,0 +1,186 @@
+
+BABE assigns a "Primary" block producer based on the VRF (normal BABE - see Randomness below) and a "Secondary" based on the following equation: `blake2_256(epoch_randomness ++ slot_number) % authorities_len`. When a block producer wins the VRF, it broadcasts a message to say that it's the primary. If no primary messages are received within some window (>>WHAT WINDOW AND HOW TO IMMUNIZE AGAINST LATENCY?), then the Secondary assignee produces the block.  With this backup in place, Polkadot slots are effectively never without blocks, with a caveat - when choosing forks, the chain with more _primary_ blocks is prioritized, so it might happen that this chain has empty slots from the VRF process.
+
+## Randomness
+
+Randomness is very important in blockchains with proof of stake. You need a way to reliably randomly select a subset of your validators to do some work. If this randomness is even the least bit predictable, then it can be abused and validators can game the system by always making sure it's their turn to be rewarded, or by attacking those they know whose turn it is in order to get them punished for unavailability.
+
+In Ethereum 2.0, randomness will come from a two-part system: the first is RANDAO, the second is VDF. RANDAO is what's called a commit-reveal scheme in which, to put it plainly, many validators secretly pick a number, and then the result is some mathematical operation performed on all those numbers, revealed one by one. Since it is not possible to predict which number all of the validators will chose, and the final output can vary wildly depending on just a single one of them, it's not possible to predict this random number. However, it is possible to influence it because the last revealer always has the option of not revealing his number, after he calculated that the current number is more in his favor than the one with his reveal. The chances of this happening are miniscule and the advantage to be gained by this bit-level influence would have to be immense - the network is plenty secure even without mitigating this risk. 
+
+However, it too will probably be mitigated by using a VDF - verifiable delay function. In layman's terms, a VDF is function which is mathematically proven to take a long time, and cannot be sped up by throwing more computers at it (it is non-parallelizable). This delayed function executes another mathematical operation on top of the RANDAO number in order to make the result come "late" - an hour and a half after the numbers have been revealed. These devices will be open source and distirbuted freely, and there is no incentive for running them. For more info on Ethereum 2 randomness, see [this post](https://our.status.im/two-point-oh-randomness/) and for more info on VDF, please see [the VDF research site](https://vdfresearch.org).
```

Comment:
This is a bit confusing, in that the article is talking about a _function_ and then pivots to a _device_.  I would add in a segue sentence indicating that the device is an ASIC which implements the VDF.